core/src/avm2/globals/Date.as

```diff
@@ -1,5 +1,461 @@
-// This is a stub - the actual class is defined in `date.rs`
 package {
-    public class Date {
+    [Ruffle(InstanceAllocator)]
+    [Ruffle(CallHandler)]
+    public dynamic class Date {
+        public static const length:int = 7;
+
+        public function Date(year:* = undefined, month:* = undefined, day:* = undefined, hours:* = undefined, minutes:* = undefined, seconds:* = undefined, ms:* = undefined) {
+            this.init(arguments);
+        }
+        private native function init(args:Array);
+
+        public static native function parse(param1:*):Number;
+
+        public static native function UTC(param1:*, param2:*, param3:* = 1, param4:* = 0, param5:* = 0, param6:* = 0, param7:* = 0, ... rest):Number;
+
+        AS3 function valueOf():Number {
+            return this.AS3::getTime();
+        }
+
+        AS3 native function toString():String;
+
+        AS3 native function toDateString():String;
+
+        AS3 native function toTimeString():String;
+
+        AS3 native function toLocaleString():String;
+
+        AS3 function toLocaleDateString():String {
+            return this.AS3::toDateString();
+        }
+
+        AS3 native function toLocaleTimeString():String;
+
+        AS3 native function toUTCString():String;
+
+        AS3 native function getUTCDay():Number;
+
+        AS3 native function getDay():Number;
+
+        AS3 native function getTimezoneOffset():Number;
+
+        AS3 native function getTime():Number;
+        AS3 native function setTime(time:* = undefined):Number;
```

Comment:
In docs/playerglobal, this isn't optional (but it is in the prototype method).

Reply:
This seems to be optional in Flash.